pkg/syncer/metadatasyncer.go

```diff
@@ -1719,6 +1721,73 @@ func pvcDeleted(obj interface{}, metadataSyncer *metadataSyncInformer) {
 	}
 }
 
+func pvAdded(obj interface{}, metadataSyncer *metadataSyncInformer) {
+	if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload && isPodVMOnStretchSupervisorFSSEnabled {
+		ctx, log := logger.GetNewContextWithLogger()
+		// Add owner reference of PV to its corresponding CNSVolumeInfo instance, if it doesn't exist.
+		pv, ok := obj.(*v1.PersistentVolume)
+		if pv == nil || !ok {
+			log.Warnf("pvAdded: unrecognized object %+v", obj)
+			return
+		}
+		log.Debugf("pvAdded: PV object %+v", pv)
+		if pv.Spec.CSI == nil {
+			log.Infof("PV %q is not a vSphere CSI volume. Skipping pvAdded functionality.", pv.Name)
```

Comment:
we need to check the driver is vSphere CSI and PV is not created by any other CSI Driver.

Reply:
Done